crypto/dh/dh_check.c

```diff
@@ -257,9 +257,27 @@ int ossl_dh_check_priv_key(const DH *dh, const BIGNUM *priv_key, int *ret)
     two_powN = BN_new();
     if (two_powN == NULL)
         return 0;
-    if (dh->params.q == NULL)
+
+    if (dh->params.q != NULL) {
+        upper = dh->params.q;
+#ifndef FIPS_MODULE
+    /*
+     * We do not have q so we just check the key is within the range
+     * we generate it.
+     */
```

Comment:
This comment is at least wrong

Reply:
Or maybe I'm confused about to what it belongs

Reply:
I've changed the comment to make it more clear as to what the check does.

Reply:
I'm confused about the comment because it belongs to the code in the `else`.